packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/Providers/ModelFactoryProvider.cs

```diff
@@ -0,0 +1,166 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Microsoft.Generator.CSharp.Expressions;
+using Microsoft.Generator.CSharp.Input;
+using Microsoft.Generator.CSharp.Primitives;
+using Microsoft.Generator.CSharp.Snippets;
+using Microsoft.Generator.CSharp.Statements;
+using static Microsoft.Generator.CSharp.Snippets.Snippet;
+
+namespace Microsoft.Generator.CSharp.Providers
+{
+    internal class ModelFactoryProvider : TypeProvider
+    {
+        private readonly IEnumerable<InputModelType> _models;
+
+        public ModelFactoryProvider(IEnumerable<InputModelType> models)
+        {
+            _models = models;
+        }
+
+        protected override string BuildName()
+        {
+            var span = CodeModelPlugin.Instance.Configuration.LibraryName.AsSpan();
+            if (span.IndexOf('.') == -1)
```

Comment:
Won't this end up scanning the whole span? Can we just do this as part of the for loop below to avoid the double scan?

Reply:
It looks like checking for the dot first in the case there is no dot and in the case there is its basically within 1 std dev

```
| Method           | NameSpace            | Mean     | Error    | StdDev   | Gen0   | Allocated |
|----------------- |--------------------- |---------:|---------:|---------:|-------:|----------:|
| RemoveWithString | FooSomething         | 21.62 ns | 0.627 ns | 0.722 ns | 0.0043 |      72 B |
| CheckForDot      | FooSomething         | 16.42 ns | 0.402 ns | 0.413 ns | 0.0042 |      72 B |
| DontCheckForDot  | FooSomething         | 35.99 ns | 0.506 ns | 0.449 ns | 0.0071 |     120 B |
| RemoveWithString | FooSomething.Bar.Baz | 61.19 ns | 0.788 ns | 0.698 ns | 0.0089 |     152 B |
| CheckForDot      | FooSomething.Bar.Baz | 47.36 ns | 0.865 ns | 0.723 ns | 0.0091 |     152 B |
| DontCheckForDot  | FooSomething.Bar.Baz | 46.40 ns | 1.921 ns | 2.212 ns | 0.0091 |     152 B |
```